tests/test_couch.py

```diff
@@ -1,54 +1,44 @@
-import unittest
-from tests.common import load_check
+from tests.common import AgentCheckTest
 from nose.plugins.attrib import attr
 from checks import AgentCheck
 
-@attr(requires='couchdb')
-class CouchDBTestCase(unittest.TestCase):
 
-    def test_metrics(self):
-        config = {
-            'instances': [{
-                'server': 'http://localhost:5984',
-            }]
-        }
-        agentConfig = {
-            'version': '0.1',
-            'api_key': 'toto'
-        }
+@attr(requires='couchdb')
+class CouchTestCase(AgentCheckTest):
 
-        self.check = load_check('couch', config, agentConfig)
+    CHECK_NAME = 'couch'
+    DB_NAMES = ['_users', '_replicator']
+    CHECK_GAUGES = [
+        'couchdb.by_db.disk_size',
+        'couchdb.by_db.doc_count',
+    ]
 
-        self.check.check(config['instances'][0])
+    def __init__(self, *args, **kwargs):
+        AgentCheckTest.__init__(self, *args, **kwargs)
+        self.config = {"instances": [{"server": "http://localhost:5984"}]}
 
-        metrics = self.check.get_metrics()
-        self.assertTrue(type(metrics) == type([]), metrics)
-        self.assertTrue(len(metrics) > 3)
-        self.assertTrue(len([k for k in metrics if "instance:http://localhost:5984" in k[3]['tags']]) > 3)
+    def test_metrics(self):
+        self.run_check(self.config)
+        for db_name in self.DB_NAMES:
+            tags = ['instance:http://localhost:5984', 'db:{0}'.format(db_name)]
+            for gauge in self.CHECK_GAUGES:
+                self.assertMetric(gauge, tags=tags, count=1)
 
     def test_service_checks(self):
-        config = {
-            'instances': [
-                {'server': 'http://localhost:5984'},
-                {'server': 'http://localhost:5985'}]
-        }
-        agentConfig = {
-            'version': '0.1',
-            'api_key': 'toto'
-        }
-
-        self.check = load_check('couch', config, agentConfig)
-        self.check.check(config['instances'][0])
-        self.assertRaises(Exception, self.check.check, config['instances'][1])
-
-        service_checks = self.check.get_service_checks()
-        self.assertEqual(len(service_checks), 2)
-
-        ok_svc_check = service_checks[0]
-        self.assertEqual(ok_svc_check['check'], self.check.SERVICE_CHECK_NAME)
-        self.assertEqual(ok_svc_check['status'], AgentCheck.OK)
-
-        cr_svc_check = service_checks[1]
-        self.assertEqual(cr_svc_check['check'], self.check.SERVICE_CHECK_NAME)
-        self.assertEqual(cr_svc_check['status'], AgentCheck.CRITICAL)
-
+        self.run_check(self.config)
+
+        self.assertServiceCheck(self.check.SERVICE_CHECK_NAME,
+                                status=AgentCheck.OK,
+                                tags=['instance:http://localhost:5984'],
+                                count=1)
```

Comment:
Nice. Can you throw a `self.coverage_report()` in the mix? It fails the test if you run the CI with `COVERAGE=true` and you didn't test all the metrics / service checks, I plan on making it a bit nicer but run it in the CI at some point.